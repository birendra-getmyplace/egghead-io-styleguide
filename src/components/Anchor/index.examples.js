import React from 'react'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import Anchor from '.'

storiesOf('Anchor', module)
  .add('default', () => (
    <Anchor url='#'>
      {lorem.words()}
    </Anchor>
  ))
