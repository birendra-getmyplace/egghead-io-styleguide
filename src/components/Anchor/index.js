import React, {PropTypes} from 'react'

const Anchor = ({
  children,
  url,
}) => (
  <a
    href={url}
    target='_blank'
    className='blue'
  >
    {children}
  </a>
)

Anchor.propTypes = {
  children: PropTypes.node.isRequired,
  url: PropTypes.string.isRequired,
}

export default Anchor
