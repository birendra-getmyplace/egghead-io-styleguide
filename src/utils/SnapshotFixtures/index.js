import React from 'react'

export const StringFixture = 'String'

export const UrlFixture = 'https://google.com'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
