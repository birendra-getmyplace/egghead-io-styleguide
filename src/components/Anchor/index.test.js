import React from 'react'
import {create} from 'react-test-renderer'
import {StringFixture, UrlFixture} from '../../utils/SnapshotFixtures'
import Anchor from '.'

test('default', () => (
  expect(create(
    <Anchor url={UrlFixture}>
      {StringFixture}
    </Anchor>
  )).toMatchSnapshot()
))
