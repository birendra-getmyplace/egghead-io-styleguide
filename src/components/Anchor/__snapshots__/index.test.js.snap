exports[`test default 1`] = `
<a
  className="blue"
  href="https://google.com"
  target="_blank">
  String
</a>
`;
